### DISCUSSION TITLE
JsonSerializerContext not working with minimalApi

### DISCUSSION THREAD

**Discussion by dany28** [dotnet/aspnetcore#52044] [General]
I've tried to switch with my microservice to Net8 Minimal API, for startup time and performance reason I've decided to use JsonSerializerContext, but I cannot achieve work solution, my simplest code for testing:
```C#
using System.Text.Json;
using System.Text.Json.Serialization;

var builder = WebApplication.CreateSlimBuilder(args);

builder.Services.ConfigureHttpJsonOptions(options =>
    options.SerializerOptions.TypeInfoResolverChain.Insert(0, AppJsonSerializerContext.Default));

var app = builder.Build();

app.MapGet("/json", () => new Person("Alex"));
app.MapGet("/string", () => JsonSerializer.Serialize(new Person("Alex"),AppJsonSerializerContext.Default.Person));

app.Run();

record Person(string FirstName)
{
}

[JsonSerializable(typeof(Person))]
internal partial class AppJsonSerializerContext : JsonSerializerContext
{

}
```
endpoint `/string` with manual serialization of course return (original field names):
```JSON
{"FirstName":"Alex"}
```
but `/json` returns result like normal serialization (camelCase field names):
```JSON
{
  "firstName": "Alex"
}
```

JsonSerializerContext not working at endpoint `/json`, resulted JSON should be the same like in `/string` end point

project configuration is default:
```XML
<Project Sdk="Microsoft.NET.Sdk.Web">

    <PropertyGroup>
        <TargetFramework>net8.0</TargetFramework>
        <Nullable>enable</Nullable>
        <ImplicitUsings>enable</ImplicitUsings>
        <InvariantGlobalization>true</InvariantGlobalization>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.0"/>
    </ItemGroup>

</Project>
```

**Comment by martincostello** [comment]
Try making this change:

```diff
  [JsonSerializable(typeof(Person))]
+ [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
  internal partial class AppJsonSerializerContext : JsonSerializerContext
  {
  }
```

_(It wasn't clear to me which one was the one you expected, so I've edited this from `Unspecified` to `CamelCase` - if you wanted the opposite, then switch the value for the other)_

**Reply by dany28** [reply]
I understand why with my configuration I've received `FirstName` with manual using `JsonSerializerContext` at endpoint `/string`, generated source contains original field names.

My question is: Why endpoint `/json` is not using `AppJsonSerializerContext` ? This line should enable it for every endpoint:
```C#
builder.Services.ConfigureHttpJsonOptions(options =>
    options.SerializerOptions.TypeInfoResolverChain.Insert(0, AppJsonSerializerContext.Default));
```

**Comment by martincostello** [comment]
Closing as this has been moved to #52088.
